dsl/pace/dsl/stencil.py

```diff
@@ -122,6 +125,148 @@ def is_gtc_backend(self) -> bool:
         return self.backend.startswith("gtc")
 
 
+def report_difference(args, kwargs, args_copy, kwargs_copy, function_name, gt_id):
+    report_head = f"comparing against numpy for func {function_name}, gt_id {gt_id}:"
+    report_segments = []
+    for i, (arg, numpy_arg) in enumerate(zip(args, args_copy)):
+        if isinstance(arg, pace.util.Quantity):
+            arg = arg.storage
+            numpy_arg = numpy_arg.storage
+        if isinstance(arg, np.ndarray):
+            report_segments.append(report_diff(arg, numpy_arg, label=f"arg {i}"))
+    for name in kwargs:
+        if isinstance(kwargs[name], pace.util.Quantity):
+            kwarg = kwargs[name].data
+            numpy_kwarg = kwargs_copy[name].data
+        else:
+            kwarg = kwargs[name]
+            numpy_kwarg = kwargs_copy[name]
+        if isinstance(kwarg, np.ndarray):
+            report_segments.append(
+                report_diff(kwarg, numpy_kwarg, label=f"kwarg {name}")
+            )
+    report_body = "".join(report_segments)
+    if len(report_body) > 0:
+        print("")  # newline
+        print(report_head + report_body)
+
+
+def compare_arr(computed_data, ref_data):
```

Comment:
Would it be helpful to move this function to `pace.util` so that we can reuse them in fv3core and physics testing too?

Reply:
It's a small function so it's fine with me to keep it as it is, but we do have three copies of the same function now in pace. I'll leave it up to you to decide.